File: src/main/java/com/icube/sim/tichu/games/hanabi/HanabiRule.java
package com.icube.sim.tichu.games.hanabi;

import com.icube.sim.tichu.games.common.domain.GameName;
import com.icube.sim.tichu.games.common.domain.GameRule;

public record HanabiRule(
        boolean isRainbowEnabled,
        RainbowMode rainbowMode,
        boolean isBlackEnabled,
        boolean isBonusEnabled
) implements GameRule {
    public HanabiRule {
        if (rainbowMode == null) {
            rainbowMode = RainbowMode.WILDCARD;
        }
    }

    public static HanabiRule createDefault() {
        return new HanabiRule(false, RainbowMode.WILDCARD, false, false);
    }

    @Override
    public GameName getGameName() {
        return GameName.HANABI;
    }

    @Override
    public int getMinPlayers() {
        return 2;
    }

    @Override
    public int getMaxPlayers() {
        return 5;
    }

    public int handSize(int playerCount) {
        return playerCount <= 3 ? 5 : 4;
    }
}

File: frontend/src/games/hanabi/domain/HanabiRule.ts
import { GameRule } from "@/games/types.ts";

export enum RainbowMode {
  WILDCARD = 'WILDCARD',
  SIXTH_LONG = 'SIXTH_LONG',
  SIXTH_SHORT = 'SIXTH_SHORT',
}

export interface HanabiRule extends GameRule {
  isRainbowEnabled: boolean;
  rainbowMode: RainbowMode;
  isBlackEnabled: boolean;
  isBonusEnabled: boolean;
}
